tpm2/tpm2_test.go

```diff
@@ -261,3 +264,128 @@ func TestHash(t *testing.T) {
 		t.Errorf("Hash(%q) returned %x, want %x", val, got, want)
 	}
 }
+
+func TestLoadExternalPublicKey(t *testing.T) {
+	pk, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rw := openTPM(t)
+	defer rw.Close()
+
+	rp := RSAParams{
+		EncAlg:     AlgRSA,
+		HashAlg:    AlgSHA1,
+		Attributes: FlagSign | FlagSensitiveDataOrigin | FlagUserWithAuth,
+		SymAlg:     AlgNull,
+		Scheme:     AlgRSASSA,
+		SchemeHash: AlgSHA1,
+		ModSize:    2048,
+		Exp:        uint32(pk.PublicKey.E),
+		Modulus:    pk.PublicKey.N.Bytes(),
+	}
+	private := Private{
+		Type:      AlgRSA,
+		Sensitive: pk.Primes[0].Bytes(),
+	}
+	h, _, err := LoadExternal(rw, rp, private, HandleNull)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer FlushContext(rw, h)
+}
+
+func TestCertify(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	params := RSAParams{
+		EncAlg:     AlgRSA,
+		HashAlg:    AlgSHA256,
+		Attributes: FlagSignerDefault,
+		SymAlg:     AlgNull,
+		Scheme:     AlgRSASSA,
+		SchemeHash: AlgSHA256,
+		ModSize:    1024,
```

Comment:
Because test code often gets copied as an example of good use, I'd prefer not to use a 1024-bit modulus here. Can this be changed to 2048?

Reply:
Done